### PR TITLE
DigiDNA Nudge updates

### LINKED ISSUE #561: com.github.macadmins.Nudge.plist huge update v1.1.0 and up
### Please provide the following information:

**Name of App / Payload**: *com.github.macadmins.Nudge.plist*

**Preference key(s) to be added / updated**: 
***User Experience*** - https://github.com/macadmins/nudge/wiki/userExperience
`allowGracePeriods`, `allowUserQuitDeferrals`, `gracePeriodInstallDelay`, `gracePeriodLaunchDelay`, `gracePeriodPath`, `showDeferralCount`
***User Interface*** - https://github.com/macadmins/nudge/wiki/userInterface
`singleQuitButton`
***OS Version Requirements*** - https://github.com/macadmins/nudge/wiki/osVersionRequirements
`actionButtonPath`, `targetedOSVersions` is now deprecated, `targetedOSVersionsRule`
***Optional Features*** - https://github.com/macadmins/nudge/wiki/optionalFeatures
`acceptableApplicationBundleIDs`, `acceptableAssertionUsage`, `acceptableAssertionApplicationNames`, `acceptableCameraUsage`, `acceptableScreenSharingUsage`, `aggressiveUserExperience`, `aggressiveUserFullScreenExperience`, `asynchronousSoftwareUpdate`, `attemptToFetchMajorUpgrade`, `attemptToBlockApplicationLaunches`, `blockedApplicationBundleIDs`, `terminateApplicationsOnLaunch`

**Applicable Preference Documentation URL(s)**:
https://github.com/macadmins/nudge/wiki/v1.1.0-features
https://github.com/macadmins/nudge/wiki/userExperience
https://github.com/macadmins/nudge/wiki/userInterface
https://github.com/macadmins/nudge/wiki/osVersionRequirements
https://github.com/macadmins/nudge/wiki/optionalFeatures

### PR DESCRIPTION
**Pull request by DigiYann**
Updated the Nudge Manifest to the latest documented preferencies
Closes #561

### CODE REVIEW

File: Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist [hunk 1]
```diff
@@ -161,60 +188,260 @@
...
+					<string>The application names using assertions which Nudge allows without taking focus. You can specify one or more applications. To find the names please run /usr/bin/pmset -g assertions in Terminal while the application is open and running.</string>
+					<key>pfm_documentation_url</key>
+					<string>https://github.com/macadmins/nudge/wiki/optionalFeatures#acceptableassertionapplicationnames---type-array-default-value--required-no</string>
+					<key>pfm_name</key>
+					<string>acceptableAssertionApplicationNames</string>
+					<key>pfm_title</key>
+					<string>Acceptable Assertion Application Names</string>
+					<key>pfm_type</key>
+					<string>array</string>
+				</dict>
```
Comment (apizz) [inline Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist:270]: We're missing a pfm_subkeys here to define this as an array of strings.

File: Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist [hunk 2]
```diff
@@ -161,60 +188,260 @@
...
+				<dict>
+					<key>pfm_app_min</key>
+					<string>1.1.7</string>
+					<key>pfm_default</key>
+					<string>false</string>
```
Comment (apizz) [inline Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist:275]: This should be a boolean false

File: Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist [hunk 3]
```diff
@@ -161,60 +188,260 @@
...
+					<key>pfm_title</key>
+					<string>Acceptable Camera Usage</string>
+					<key>pfm_type</key>
+					<string>boolean</string>
+				</dict>
+				<dict>
+					<key>pfm_app_min</key>
+					<string>1.1.7</string>
+					<key>pfm_default</key>
+					<string>false</string>
```
Comment (apizz) [inline Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist:293]: boolean false

**Review by apizz — CHANGES_REQUESTED** [review]
I'm not sure if you have the pre-commit config installed because this and a few other things should have been caught automatically. 
 Can you please install these? 
 https://github.com/ProfileCreator/ProfileManifests/wiki/Contribute-to-an-Existing-Manifest#2-install--enable-pre-commit

File: Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist [hunk 4]
```diff
@@ -161,60 +188,271 @@
...
+
+Disables Nudge's re-activation logic for the listed applications. If one of the specified applications is foremost, Nudge will not reactivate itself. (This feature does not work for Nudge's initial activation.).</string>
+					<key>pfm_documentation_url</key>
+					<string>https://github.com/macadmins/nudge/wiki/optionalFeatures#acceptableapplicationbundleids---type-array-default-value--required-no</string>
+					<key>pfm_name</key>
+					<string>acceptableApplicationBundleIDs</string>
+					<key>pfm_subkeys</key>
+					<array>
+						<dict>
+							<key>pfm_name</key>
```
Comment (apizz) [inline Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist:230]: As this is contained an array, the key name here should actually be `pfm_title`.  `pfm_name` is only used when we have a `<key></key>` and value, and as such wouldn't be in the case of these strings w/in the array.

File: Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist [hunk 5]
```diff
@@ -161,60 +188,271 @@
...
+					<string>acceptableAssertionApplicationNames</string>
+					<key>pfm_title</key>
+					<string>Acceptable Assertion Application Names</string>
+					<key>pfm_subkeys</key>
+  					<array>
+    					<dict>
+							<key>pfm_name</key>
+							<string>acceptableAssertionApplicationName</string>
+							<key>pfm_title</key>
+							<string>Acceptable Assertion Application Name</string>
```
Comment (apizz) [inline Manifests/ManagedPreferencesApplications/com.github.macadmins.Nudge.plist:274]: Same as previous comment, no `pfm_name` should be here when the main preference is an array.  `pfm_title` is all that's needed besides `pfm_type`

**Review by apizz — CHANGES_REQUESTED** [review]
I think you're still running into issues because there are a number of errors that _would_ be caught if you had installed the `pre-commit` checks.  Please do this for next time, but I'm going through this now to help get across the finish line

**Comment by DigiYann** [comment]
Hi AP, thank you for sharing the tool. I have installed it and it rocks. So, do you want to push your correction or should I? I guess that you do following what you wrote (Please do this for next time, but I'm going through this now to help get across the finish line) but keep me posted. Thank you. Best, Y

**Comment by apizz** [comment]
I forget that your fork branches are protected so I can't contribute back to them to collaborate on PRs like this 🙁 

So if you have the pre-commit setup installed and you bump the last modified date for example it should yell at you with the mis-matched `pfm_default` values.

Lines 245, 320, 336, 352, 368, 404, 436, 683, & 699 should not be `<string>true|false</string>` they should just be either `<true/>` or `<false/>`

**Comment by apizz** [comment]
That's also in addition to the other comments I made RE not using `pfm_name` in `pfm_subkeys` where the 'parent' preference is an array and subkey is a string

**Review by apizz — APPROVED** [review]

**Comment by apizz** [comment]
Just moving this across the finish line

**Pull request merged**